src/taxonomy/taxonomy-detail/TaxonomyDetailPage.jsx

```diff
@@ -51,32 +76,35 @@ const TaxonomyDetailPage = () => {
     />
   );
 
-  const onClickMenuItem = (menuName) => {
-    switch (menuName) {
-    case 'export':
-      setIsExportModalOpen(true);
-      break;
-    default:
-      break;
-    }
-  };
-
-  const getHeaderActions = () => (
-    <TaxonomyDetailMenu
-      id={taxonomy.id}
-      name={taxonomy.name}
-      disabled={
-        // We don't show the export menu, because the system-taxonomies
-        // can't be exported. The API returns and error.
-        // The entire menu has been disabled because currently only
-        // the export menu exists.
-        // ToDo: When adding more menus, change this logic to hide only the export menu.
-        taxonomy.systemDefined
-      }
-      onClickMenuItem={onClickMenuItem}
+  const renderDeleteDialog = () => isDeleteDialogOpen && (
+    <DeleteDialog
+      isOpen={isDeleteDialogOpen}
+      onClose={() => setIsDeleteDialogOpen(false)}
+      onDelete={onClickDeleteTaxonomy}
+      taxonomyName={taxonomy.name}
+      tagsCount={0}
     />
   );
 
+  const onClickMenuItem = (menuName) => (
+    menuItemActions[menuName]?.()
+  );
+
+  const getHeaderActions = () => {
+    let enabledMenuItems = menuItems;
+    if (taxonomy.systemDefined) {
+      enabledMenuItems = systemDefinedMenuItems;
+    }
```

Comment:
Don't we also need to check whether the user is allowed to delete this taxonomy before we show 'Delete' as a menu option?